azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -432,9 +443,19 @@ def _log_error_if_none(attr_name, value):
         if ext_handler.properties.state in (None, ""):
             raise ExtensionConfigError("Received empty Extensions.Plugins.Plugin.state, failing Handler")
 
-        location = getattrib(plugin, "location")
-        failover_location = getattrib(plugin, "failoverlocation")
-        for uri in [location, failover_location]:
+        def getattrib_wrapped_in_list(node, attr_name):
+            attr = getattrib(node, attr_name)
+            return [attr] if attr not in (None, "") else []
+
+        location = getattrib_wrapped_in_list(plugin, "location")
+        failover_location = getattrib_wrapped_in_list(plugin, "failoverlocation")
+
+        additional_location_node = find(plugin, "additionalLocations")
```

Comment:
`In regions with this feature enabled,
        CRP will provide any extra links in the format above. If no extra links
        are provided, the subnode will not exist.
`
if the subnode does not exit, find will return None. Let's add a check for that

Reply:
`findall` actually handles this on testutil.py#45:
```python
if root is None:
    return []
```

I suppose we could inline the variable (so that, in the future, changes don't use `additional_location_node` thinking it is non-None), but current functionality isn't effected by this (when I tested this PR in the wrong region, the node was missing and things worked as expected).

Reply:
Sure, the current code handles None, but the fact that this variable can be None is not clear from the code, and as this code changes bugs may be introduced.

Being more explicit would help

```
additional_location_node = find(plugin, "additionalLocations")
if additional_location_node is not None:
   nodes_list = findall(additional_location_node, "additionalLocation")
   additional_locations = [gettext(node) for node in nodes_list]
   locations = location + additional_locations + failover_location
```